pkg/informer/informer_test.go

```diff
@@ -126,6 +126,10 @@ func Test_listAndResync_syncWithPrevState(t *testing.T) {
 	if *deleteFuncCount != 1 {
 		t.Errorf("Expected informer.DeleteFunc to be called 1 time, but got %d.", *deleteFuncCount)
 	}
+	// Verify the resources no longer exist in informer.resourceIndex:
+	if _, ok := informer.resourceIndex["fake-uid"]; ok {
+		t.Errorf("Expected informer.resourceIndex to be empty")
+	}
```

Comment:
A better test would be to validate that `inform.DeleteFunc` gets called with the correct args.

Reply:
Actually, we need a new test where we stop the informer and validate that  `inform.DeleteFunc` is called with the expected data.